Automattic/WP-Job-Manager
# Add option to determine how to handle shortcodes in job listings

Fixes #1146 

### Changes Proposed in this Pull Request

* Add option to determine how to handle shortcodes in job listings

### Testing Instructions

Using this branch of WPJM:

1. Create a new job listing
2. Use a valid shortcode on the description of that job listing, you can for instance use this as post content:

```
Test

[job_apply id=”X”]

Another Test
```
3. Check the job listing on the frontend (the individual job listing page), make sure the shortcode runs as usual
4. Now, on wp-admin, go to "Job Manager" -> "Settings" -> "Job Listings" and check the option "Strip Shortcodes" (that option should be disabled by default)
5. Go back to the individual job listing page for the job listing you just created and make sure the shortcode DOESN'T RUN at all, like, make sure it is actually stripped from the page.


### Release Notes

- Add option to strip shortcodes from the job description

### Screenshot / Video

![New option to strip shortcodes](https://github.com/Automattic/WP-Job-Manager/assets/529864/8f01b218-0f7f-4f5a-9b4f-756c24f058de)

### Notes

Just for the sake of documentation, initially I was going to add an option to Allow Shortcodes (the current default), Keep them (by just not running them at all) or Strip them (which is what I'm doing here). This was done until commit 61ca9ae8cfb11faac1334713af38484bb6fbfe48, so if you want to check out that approach too, you can. I decided to simplify things because the option to Keep the shortcodes was a bit...potentially troublesome, in special because it might not work because of third-party plugins/themes. So just allowing to strip them sounded easier.


----

| Plugin build for 744a58e56eeb9ace2a79570cd88f23f0ac9e36ed <a href="#"><img width=600></a> |
| ------------------------------------------------------------ |
| 📦 [Download plugin zip](https://wpjobmanager.com/wp-content/uploads/2023/11/wp-job-manager-zip-2662-744a58e5.zip)                       |
| ▶️ [Open in playground](https://wpjobmanager.com/playground/?core=2023/11/2662-744a58e5)             |
